File: ios/calorietracker/Models/SupplementalNutrient.swift
import Foundation

enum SupplementalNutrient: String, CaseIterable, Identifiable, Codable {
    case creatine
    case betaAlanine
    case lCitrulline
    case lCarnitine
    case lArginine
    case taurine
    case betaine
    case hmb

    var id: String { rawValue }

    var jsonKey: String {
        switch self {
        case .creatine: "creatine"
        case .betaAlanine: "beta_alanine"
        case .lCitrulline: "l_citrulline"
        case .lCarnitine: "l_carnitine"
        case .lArginine: "l_arginine"
        case .taurine: "taurine"
        case .betaine: "betaine"
        case .hmb: "hmb"
        }
    }

    var displayName: String {
        switch self {
        case .creatine: "Creatine"
        case .betaAlanine: "Beta-Alanine"
        case .lCitrulline: "L-Citrulline"
        case .lCarnitine: "L-Carnitine"
        case .lArginine: "L-Arginine"
        case .taurine: "Taurine"
        case .betaine: "Betaine"
        case .hmb: "HMB"
        }
    }

    var optionalNutrient: OptionalNutrient {
        OptionalNutrient(rawValue: rawValue)!
    }
}

File: android/app/src/main/java/com/ayuvo/health/models/SupplementalNutrient.kt
package com.ayuvo.health.models

import com.ayuvo.health.R

enum class SupplementalNutrient(
    val storageKey: String,
    val apiKey: String,
    val displayName: String,
    val displayNameRes: Int
) {
    CREATINE("creatine", "creatine", "Creatine", R.string.nutrition_label_creatine),
    BETA_ALANINE("betaAlanine", "beta_alanine", "Beta-Alanine", R.string.nutrition_label_beta_alanine),
    L_CITRULLINE("lCitrulline", "l_citrulline", "L-Citrulline", R.string.nutrition_label_l_citrulline),
    L_CARNITINE("lCarnitine", "l_carnitine", "L-Carnitine", R.string.nutrition_label_l_carnitine),
    L_ARGININE("lArginine", "l_arginine", "L-Arginine", R.string.nutrition_label_l_arginine),
    TAURINE("taurine", "taurine", "Taurine", R.string.nutrition_label_taurine),
    BETAINE("betaine", "betaine", "Betaine", R.string.nutrition_label_betaine),
    HMB("hmb", "hmb", "HMB", R.string.nutrition_label_hmb)

    ;

    val optionalNutrient: OptionalNutrient
        get() = OptionalNutrient.valueOf(name)
}
